### PR TITLE
OCT-522 - Approvals Tracker

### PR DESCRIPTION
**Pull request by Harry-Jisc**
The purpose of this PR was create the approvals tracker for authors to understand where their publications stands in relation to being published.

---

### Acceptance Criteria:
As per [OCT-522](https://jiscdev.atlassian.net/browse/OCT-522?atlOrigin=eyJpIjoiZTFlZGJhZDY3NGUwNDJhNGJlYjdkY2NiZWU2MDU4ZmQiLCJwIjoiaiJ9)

[OCT-522]: https://jiscdev.atlassian.net/browse/OCT-522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Harry-Jisc** [comment]
There is more that could be tested in this wit the e2e test, however it's much simpler to test once [OCT-305](https://github.com/JiscSD/octopus/pull/333) has been merged due to simplicity of accessing co-authored publications. 

This is not blocked by that, but the testing could be more extensive if it included that. I will mark as ready for review and we can put back into draft if seen as requiring the additional testing.

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [inline ui/src/components/ApprovalsTracker/index.tsx:14]: Based on the mockup provided, the font should be semi-bold here

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [inline ui/src/components/ApprovalsTracker/index.tsx:18]: There should be more space between the table and the publication title

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [inline ui/src/components/ApprovalsTracker/index.tsx:41]: *Corresponding

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [reply ui/src/components/ApprovalsTracker/index.tsx:18]: On mobile devices, if the email address is long enough, there's no way to see 'Status' column
image

Horizontal scroll should be available if the content overflows

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [reply ui/src/components/ApprovalsTracker/index.tsx:41]: 'Corresponding Author' is displayed on the 'Status' column, we don't need to display it here also

Comment (florin-holhos) [reply ui/src/components/ApprovalsTracker/index.tsx:41]: On the mockup, there's a small "(you)" text next to the current logged in user name
image

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [inline ui/src/components/ApprovalsTracker/index.tsx:7]: 'publicationData' prop includes everything we need to display on the approvals tracker
Do we really need the 'user' prop?

File: ui/src/components/ApprovalsTracker/index.tsx
Comment (florin-holhos) [inline ui/src/components/ApprovalsTracker/index.tsx:45]: On line 39 you are checking  the creator against an author in the coAuthor array.
Here, you are checking the same thing, but against the 'user' prop.

File: ui/src/pages/publications/[id]/index.tsx
Comment (florin-holhos) [inline ui/src/pages/publications/[id]/index.tsx:279]: In the mockup, this banner also says: "All authors must approve this draft before it can be published."

**Review by florin-holhos — APPROVED** [review]

File: ui/src/components/index.tsx
Comment (florin-holhos) [inline ui/src/components/index.tsx:91]: forgot to remove this import

File: ui/src/pages/publications/[id]/index.tsx
Comment (florin-holhos) [inline ui/src/pages/publications/[id]/index.tsx:16]: forgot to remove this import. ApprovalsTracker was added to /components exports

**Review by florin-holhos — APPROVED** [review]

**Review by florin-holhos — APPROVED** [review]

**Pull request merged**
